controller/appcontroller.go

```diff
@@ -139,6 +139,7 @@ func NewApplicationController(
 	persistResourceHealth bool,
 	clusterFilter func(cluster *appv1.Cluster) bool,
 	applicationNamespaces []string,
+	shard string,
```

Comment:
let's rename this to `shardingAlgorithm`. shard does not reflect what it is referring to.